c++/src/capnp/compat/http-over-capnp.c++

```diff
@@ -693,12 +690,30 @@ public:
   kj::StringPtr url;
   kj::HttpHeaders headers;
   capnp::HttpService::ClientRequestContext::Client clientContext;
-  kj::Own<kj::PromiseFulfiller<kj::Promise<void>>> replyFulfiller;
+  bool responseSent = false;
 
   static kj::HttpMethod validateMethod(capnp::HttpMethod method) {
     KJ_REQUIRE(method <= capnp::HttpMethod::UNSUBSCRIBE, "unknown method", method);
     return static_cast<kj::HttpMethod>(method);
   }
+
+  template <typename T>
+  void dontWaitForRpc(kj::Promise<T> promise) {
+    // When we call clientContext.startResponse(), we really don't want to actually wait for
+    // the reply to this RPC, because in many cases we will call this, write a response body,
+    // and then immediately return from the HttpService::requset() handler. At that point, we
+    // would like CapnpToKjHttpServiceAdapter::request() to be able to propagate this return
+    // immediately *without* waiting for a round trip to the client, so without waiting for
+    // `startResponse()` to finish. However, we also do not want to inadvertently cancel
+    // `startResponse()`, so we have to save the promise somewhere. Since this is an RPC that
+    // doesn't depend on the lifetime of any other object... we will just detach() it.
+
+    promise.detach([](kj::Exception&& e) {
+      if (e.getType() == kj::Exception::Type::FAILED) {
```

Comment:
I'm used to seeing `!= DISCONNECTED` to inhibit logging, so this stuck out to me. I guess OVERLOADED isn't likely to occur, and we wouldn't want to log REMOTE exceptions?

Reply:
There's no REMOTE type. I figured OVERLOADED is probably not relevant to log, and UNIMPLEMENTED isn't going to be an issue here. Really I could have skipped logging any exceptions here since presumably the problem is going to be noticed elsewhere but figured logging FAILED is pretty safe since it probably represents a real problem.